README.md

```diff
@@ -0,0 +1,80 @@
+# TextValue <GitHub path="RedMadRobot/textvalue/tree/main/"/>
+[![Version](https://img.shields.io/maven-central/v/com.redmadrobot.textvalue/textvalue?style=flat-square)][mavenCentral]
+[![License](https://img.shields.io/github/license/RedMadRobot/textvalue?style=flat-square)][license]
+
+TextValue is an abstraction allowing to work with a `String` and a string resource ID the same way.
+
+---
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Installation](#installation)
+- [Usage](#usage)
+- [Contributing](#contributing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Installation
+
+Add the dependency:
+```groovy
+repositories {
+    mavenCentral()
+    google()
+}
+
+dependencies {
+    // Views version
+    implementation("com.redmadrobot.textvalue:textvalue:1.0.0")
+
+    // Compose extensions for textvalue
+    implementation("com.redmadrobot.textvalue:textvalue-compose:1.0.0")
+}
+```
+
+## Usage
+
+**TextValue** is a wrapper to make it possible to work with plain `String` and `StringRes` in the same way.
+It may be useful for cases when you want to fallback to `StringRes` if desired string value is `null`.
+
+You can wrap `String` and `StringRes` with `TextValue` using `TextValue(String)`, `TextValue(Int)` or `TextValue(String?, Int))`, and use method `TextValue.get(Resource)` to retrieve `String`:
+
+```kotlin
+// in some place where we can't access Context
+val errorMessage = TextValue(exception.message, defaultResourceId = R.string.unknown_error)
+showMessage(errorMessage)
+
+// in Activity, Fragment or View
+fun showMessage(text: TextValue) {
+    val messageText = text.get(resources)
+    //...
+}
+```
+
+`TextValue` also support `Jetpack compose`:
+
+```kotlin
+// in Composable functions
+@Composable
+fun Screen(title: TextValue) {
+    val recource = LocalContext.current.resources
+    Text(text = title.get(recource))
```

Comment:
I think we should only keep variant with `stringResource` to promote best practices